src/serde/deserialize_utils.rs

```diff
@@ -41,185 +54,139 @@ pub fn maybe_add_padding(mut hex: String) -> String {
     hex
 }
 
-fn parse_register(splitted_value_str: &[&str]) -> Option<Register> {
-    let str_tmp: Vec<&str> = splitted_value_str[0].split(',').collect();
-
-    let mut raw_reg_str = str_tmp[0].to_string();
-
-    raw_reg_str.retain(|c| !r#"["#.contains(c));
+/* NOM PARSERS */
```

Comment:
Could you document the grammar being parsed (just an overview) in the dev docs and link to the documentation in the comments at the beginning of the parser?

Reply:
Will be done afterwards #340